agent/eventhandler/task_handler_types.go

```diff
@@ -28,9 +29,13 @@ type sendableEvent struct {
 
 	taskSent   bool
 	taskChange api.TaskStateChange
+
+	mu sync.RWMutex
```

Comment:
Yep. Its protecting the state for the whole struct. If `mu` is too short for our conventions I can do `lock` , `eventLock` , `lockAllOfTheThings` (in ascending order of personal preference).